src/game/behaviors/blue_fish.inc.c

```diff
@@ -1,76 +1,131 @@
-            o->oForwardVel = o->oBlueFishUnkF4 + 3.0f;
-            if (o->oTimer >= o->oBlueFishUnkF8 + 60)
+            
+            // Set forward velocity and progress oAction
+            o->oForwardVel = o->oBlueFishRandomVel + 3.0f;
+            if (o->oTimer >= o->oBlueFishRandomTime + 60) {
                 o->oAction++;
-            if (o->oTimer < (o->oBlueFishUnkF8 + 60) / 2)
+            }
+            
+            // Set pitch velocity
+            if (o->oTimer < (o->oBlueFishRandomTime + 60) / 2) {
                 o->oFaceAnglePitch += o->oAngleVelPitch;
-            else
+            } else {
                 o->oFaceAnglePitch -= o->oAngleVelPitch;
+            }
+            
+            // Calculate new Y velocity
             o->oVelY = -sins(o->oFaceAnglePitch) * o->oForwardVel;
             break;
-        case 1:
+        
+        // Animates and adjusts fish yaw angle, then progresses oAction.
+        case BLUE_FISH_RANDOM_YAW:
             func_8029ED98(0, 2.0f);
-            o->oMoveAngleYaw = (s32)(o->oBlueFishUnk100 + o->oMoveAngleYaw);
-            if (o->oTimer == 15)
+            o->oMoveAngleYaw = (s32)(o->oBlueFishRandomAngle + o->oMoveAngleYaw);
+            if (o->oTimer == 15) {
                 o->oAction++;
+            }
             break;
-        case 2:
+            
+        // Animates and adjusts pitch velocity.
+        case BLUE_FISH_RANDOM_PITCH:
             func_8029ED98(0, 1.0f);
-            if (o->oTimer >= o->oBlueFishUnkF8 + 60)
+            
+            // Progresses oActioon after some time. 
+            if (o->oTimer >= o->oBlueFishRandomTime + 60) {
                 o->oAction++;
-            if (o->oTimer < (o->oBlueFishUnkF8 + 60) / 2)
+            }
+            
+            // Adjusts pitch angle. Direction relies on time not passed.
+            if (o->oTimer < (o->oBlueFishRandomTime + 60) / 2) {
                 o->oFaceAnglePitch -= o->oAngleVelPitch;
-            else
+            } else {
                 o->oFaceAnglePitch += o->oAngleVelPitch;
+            }
             break;
-        case 3:
+            
+        // Animates and adjusts yaw angle
+        case BLUE_FISH_RANDOM_YAW_2:
             func_8029ED98(0, 2.0f);
-            o->oMoveAngleYaw = (s32)(o->oBlueFishUnk100 + o->oMoveAngleYaw);
-            if (o->oTimer == 15)
-                o->oAction = 0;
+            o->oMoveAngleYaw = (s32)(o->oBlueFishRandomAngle + o->oMoveAngleYaw);
+            
+            // Sets oAction to the spawn phase if the timer equals fifteen.
+            if (o->oTimer == 15) {
+                o->oAction = BLUE_FISH_SPAWN;
+            }
             break;
     }
+    
+    // Calculates Y velocity and calls physics engine.
     o->oVelY = -sins(o->oFaceAnglePitch) * o->oForwardVel;
     obj_move_using_fvel_and_gravity();
-    if (o->parentObj->oAction == 2)
+    
+    // Deletes object if the parent has oAction set to two.
+    if (o->parentObj->oAction == 2) {
```

Comment:
Label oAction here, isn't the parent object the fish group which is in the same file? GitHub reviewing system doesn't let me see the whole file lol

Reply:
3 dots -> view file

I do not know what the parentObject is.

Reply:
then investigate!